cmd/deploy/proxy.go

```diff
@@ -71,10 +73,16 @@ type proxyHandler struct {
 }
 
 // NewProxy creates a new proxy
-func NewProxy(kubeconfig *KubeConfig) (*Proxy, error) {
+func NewProxy(kubeconfig kubeConfigHandler, portGetter func(iface string) (int, error)) (*Proxy, error) {
 	// Look for a free local port to start the proxy
-	port, err := model.GetAvailablePort("localhost")
+	port, err := portGetter("localhost")
 	if err != nil {
+		if dnsError, ok := err.(*net.DNSError); ok && dnsError.IsNotFound {
+			return nil, oktetoErrors.UserError{
+				E:    fmt.Errorf("could not find available ports: %w", dnsError),
```

Comment:
I think adding this context to the error might be confusing. If I face the error 
<img width="910" alt="Captura de Pantalla 2023-10-18 a las 10 47 21" src="https://github.com/okteto/okteto/assets/22500940/9a26c15f-507a-40a5-a5d7-f40e5be82728">
I could think that there is some problem with some ports but it is not the case. Adding this context to the original error seems weird for me in every case. I would return just the error that comes from `portGetter` WDYT?

Reply:
i think that the function throwing the error is to get the available ports on a host, so it kind on makes sense to me. i initially just returned the error but this was modified after the revision of nacho.